### DISCUSSION TITLE
Slight Hiccup With the Zig-based Client

### DISCUSSION THREAD

**Discussion by tonytins** [tonytins/mysimulation#8] [Announcements]
Turns out, even if I got the rendering figured out, there isn't much in the way of proper HTTP client libraries. xD Only server frameworks. Nor does the current Raylib Zig bindings include the UI module. I know Zig will get there eventually, but for the time being, I'm going to put that version on the back burner.

However, in order to keep things on track, I'm going to switch to C# because I'm most comfortable with that language. I know Rust is way, way more ideal in terms of performance, but I don't want to deal with large, GB-sized projects, especially for a game of this magnitude. I can't stress enough that this is **only an experiment**. So, please forgive me.

**Comment by JipBoesenkool** [comment]
Zig can consume C/C++ as long as it is respects ABI.
I was thinking of writing the render layer in C++. 
This would reduce the bindings to dependencies such as ray and if properly exposed should be consumable by any language,

**Reply by tonytins** [reply]
My biggest problem is the networking. Which I was hoping to at least test, but I guess I was a little ahead of myself.

**Reply by JipBoesenkool** [reply]
Can't we do the same with networking?
